src/wallet/wallet.cpp

```diff
@@ -1894,6 +1894,30 @@ TransactionError CWallet::FillPSBT(PartiallySignedTransaction& psbtx, bool& comp
         }
     }
 
+    // Figure out if any witness_utxos should be dropped
```

Comment:
833ecb8dec1a10e3bc1b8d5bcaf0b6839277f582: `non_witness_utxos` (also below)

Don't do this for `SIGHASH_SINGLE` and `SIGNHASH_ANYONECANPAY`, since new non-taproot inputs might be added later? Or should whoever adds such an input just add them back in? Maybe add a functional test for this.

Reply:
It only matters for `SIGHASH_ANYONECANPAY`. I've added a check for that.